data_model/src/permission.rs

```diff
@@ -17,196 +18,113 @@ pub type Permissions = BTreeSet<Permission>;
-        Self {
-            definition_id,
-            payload: JsonString(payload.to_owned()),
-        }
-    }
-
-    /// Construct [`Self`]
-    pub fn new(definition_id: PermissionId, payload: &serde_json::Value) -> Self {
+    /// Constructor
+    pub fn new(id: PermissionId, payload: impl IntoJsonString) -> Self {
         Self {
-            definition_id,
-            payload: JsonString::new(payload),
+            id,
+            payload: payload.into_json_string(),
         }
     }
-
-    /// Return id of this token
-    // TODO: Use getset to derive this after fixes in FFI
-    pub fn definition_id(&self) -> &Name {
-        &self.definition_id
-    }
-
-    /// Payload of this token
-    // TODO: Use getset to derive this after fixes in FFI
-    pub fn payload(&self) -> &String {
-        &self.payload.0
-    }
-}
-
-impl core::fmt::Display for Permission {
-    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
-        write!(f, "{}", self.definition_id)
-    }
-}
-
-impl JsonString {
-    /// Construct [`JsonString`]
-    pub fn new(payload: &serde_json::Value) -> Self {
-        Self(payload.to_string())
-    }
-}
-impl PartialEq for JsonString {
-    fn eq(&self, other: &Self) -> bool {
-        serde_json::from_str::<serde_json::Value>(&self.0).unwrap()
-            == serde_json::from_str::<serde_json::Value>(&other.0).unwrap()
-    }
-}
-
-impl<'de> serde::de::Deserialize<'de> for JsonString {
-    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
-    where
-        D: serde::Deserializer<'de>,
-    {
-        let json = serde_json::Value::deserialize(deserializer)?;
-        Ok(Self::new(&json))
-    }
 }
 
-impl serde::ser::Serialize for JsonString {
-    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
-    where
-        S: serde::Serializer,
-    {
-        let json = serde_json::Value::from_str(&self.0).map_err(serde::ser::Error::custom)?;
-        json.serialize(serializer)
+impl Borrow<str> for PermissionId {
+    fn borrow(&self) -> &str {
+        self.name.borrow()
     }
 }
 
-impl iroha_schema::TypeId for JsonString {
-    fn id() -> iroha_schema::Ident {
-        "JsonString".to_owned()
+impl Borrow<str> for Permission {
+    fn borrow(&self) -> &str {
+        self.id.borrow()
     }
 }
 
-impl IntoSchema for JsonString {
-    fn type_name() -> iroha_schema::Ident {
-        <Self as iroha_schema::TypeId>::id()
-    }
-
-    fn update_schema_map(map: &mut iroha_schema::MetaMap) {
-        if !map.contains_key::<Self>() {
-            map.insert::<Self>(iroha_schema::Metadata::String);
-        }
+impl From<Name> for PermissionId {
+    fn from(name: Name) -> Self {
+        Self { name }
     }
 }
 
-impl PartialOrd for JsonString {
-    fn partial_cmp(&self, other: &Self) -> Option<core::cmp::Ordering> {
-        Some(self.cmp(other))
+impl From<PermissionId> for Name {
+    fn from(value: PermissionId) -> Self {
+        value.name
     }
 }
```

Comment:
are these impls used somewhere? We don't usually assume that id can be converted to name or str

Reply:
Cleaned these.